lib/block.ml

```diff
@@ -145,41 +144,44 @@ let dump ppf ({ loc; section; labels; contents; value; _ } as b) =
     labels
     Fmt.(Dump.option Header.pp)
     (header b)
-    Fmt.(Dump.list dump_string)
+    Fmt.Dump.(list string)
     contents dump_value value
 
-let pp_lines syntax t =
-  let pp =
-    match syntax with
-    | Some Syntax.Cram -> Fmt.fmt "  %s"
-    | Some Syntax.Mli ->
-        fun ppf -> Fmt.fmt "%*s%s" ppf (t.loc.loc_start.pos_cnum + 2) ""
-    | _ -> Fmt.string
+let pp_lines syntax _t ppf lines =
+  let print_line ppf first last line =
+    (match syntax with
+    | Some Syntax.Cram -> Fmt.string ppf line
+    | Some Syntax.Mli -> (
+        match (first, last) with
+        | true, _ | _, true -> Fmt.string ppf (String.trim line)
+        | false, false -> Fmt.string ppf line)
+    | Some Syntax.Normal | None -> Fmt.string ppf line);
+    match last with false -> Fmt.pf ppf "\n" | true -> ()
```

Comment:
I am not sure why this trimming on first and last lines in mli files. in particular, with this:
```
{[             
   "11" ]}
```
is tranformed like this (there are removed whitespace on the first line):
```
-{[             
-   "11" ]}
+{[
+"11"]}
```

I suggest:
```suggestion
  let print_line ppf first last line =
    Fmt.string ppf line;
    match last with false -> Fmt.pf ppf "\n" | true -> ()
```
which however require changes elsewhere as it breaks the tests: something must be adding padding for last line!